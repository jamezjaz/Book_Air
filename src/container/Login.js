import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import login from '../apiRequests/loginRequest';

const Login = props => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errorMsg, setErrorMsg] = useState('');

  const handleLogin = () => {
    const { history } = props;
    history.push('/airlinesList');
  };

  const handleChange = event => {
    if (event.target.name === 'email') {
      setEmail(event.target.value);
    } else {
      setPassword(event.target.value);
    }
    // console.log(event.target);
  };

  const handleSubmit = event => {
    event.preventDefault();
    const { signIn } = props;
    const user = {
      email,
      password,
    };
    if (email || password !== '') {
      setErrorMsg('');
      signIn(user);
      handleLogin();
    } else {
      setErrorMsg('Please, enter correct credentials');
    }
    console.log(user);
  };

  return (
    <div>
      <form>
        <label htmlFor="email">
          Email:
          <input type="text" name="email" id="email" placeholder="Enter your email" onChange={handleChange} value={email.email} />
        </label>
        <br />
        <label htmlFor="password">
          Password:
          <input type="password" name="password" id="password" placeholder="Enter your password" onChange={handleChange} value={password.password} />
        </label>
        <br />
        <button type="submit" onClick={handleSubmit}>Login</button>
        {errorMsg === '' ? '' : <h3>{errorMsg}</h3>}
      </form>
    </div>
  );
};

Login.propTypes = {
  signIn: PropTypes.func.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
};

const mapDispatchToprops = dispatch => ({
  signIn: user => dispatch(login(user)),
});

export default connect(null, mapDispatchToprops)(Login);
