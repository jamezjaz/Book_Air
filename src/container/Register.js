import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import registration from '../apiRequests/regRequest';

const Register = props => {
  const user = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleChange = event => {
    if (event.target.name === 'name') {
      setName(event.target.value);
    } else if (event.target.name === 'email') {
      setEmail(event.target.value);
    } else {
      setPassword(event.target.value);
    }
    // console.log(event.target);
  };

  const handleRegister = () => {
    const { history } = props;
    history.push('/airlinesList');
  };

  const handleSubmit = event => {
    event.preventDefault();
    console.log('Initial User', user);
    dispatch(registration({
      name,
      email,
      password,
    }));
    handleRegister();
  };

  return (
    <div>
      <form>
        <label htmlFor="name">
          Name:
          <input type="text" name="name" id="name" placeholder="Enter your name" onChange={handleChange} value={name.name} />
        </label>
        <br />
        <label htmlFor="email">
          Email:
          <input type="text" name="email" id="email" placeholder="Enter your email" onChange={handleChange} value={email.email} />
        </label>
        <br />
        <label htmlFor="password">
          Password:
          <input type="password" name="password" id="password" placeholder="Enter your password" onChange={handleChange} value={password.password} />
        </label>
        <br />
        <button type="submit" onClick={handleSubmit}>Register</button>
      </form>
    </div>
  );
};

Register.propTypes = {
  history: PropTypes.instanceOf(Object).isRequired,
};

export default Register;
