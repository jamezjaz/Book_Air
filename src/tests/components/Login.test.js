import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import Register from '../../container/Register';
import Login from '../../container/Login';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Login', () => {
  test('matches Login snapshot', () => {
    const tree = renderer
      .create(
        (
          <Provider store={store}>
            <Router>
              <Login />
            </Router>
          </Provider>
        ),
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Register component', () => {
    const { queryAllByTestId } = render(
      (
        <Provider store={store}>
          <Router>
            <Route path="/register" component={Register} />
          </Router>
        </Provider>
      ),
    );
    const home = queryAllByTestId('home');
    expect(home).toBeTruthy();
  });

  test('it should have a Login form', () => {
    const { queryAllByTestId } = render(
      (
        <Provider store={store}>
          <Router>
            <Route path="/register" component={Register} />
          </Router>
        </Provider>
      ),
    );
    const form = queryAllByTestId('form');
    expect(form).toBeTruthy();
  });
});
