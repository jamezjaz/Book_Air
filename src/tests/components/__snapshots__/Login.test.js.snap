// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches Login snapshot 1`] = `
<div
  className="container-fluid"
>
  <a
    className="d-flex justify-content-end"
    href="/"
    onClick={[Function]}
  >
    <button
      data-testid="home"
      type="button"
    >
      Home
    </button>
  </a>
  <form
    data-testid="form"
    name="myForm"
    onSubmit={[Function]}
  >
    <label
      className="text-left"
      htmlFor="email"
    >
      Email:
      <br />
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="password"
    >
      Password:
      <br />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </label>
    <br />
    <button
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
    <div>
      <span>
        Don't have an account? 
      </span>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
    
  </form>
</div>
`;
